walkeeper/src/replication.rs

```diff
@@ -204,18 +210,27 @@ impl ReplicationConn {
         let mut wal_file: Option<File> = None;
 
         loop {
-            /* Wait until we have some data to stream */
-            if let Some(lsn) = swh.timeline.get().wait_for_lsn(start_pos) {
-                end_pos = lsn
+            /* For recovery walproposer needs all WAL safekeeper has whether committed or not */
+            if stop_pos != Lsn(0) {
+                if start_pos >= stop_pos {
+                    /* recovery finished */
```

Comment:
Nit: I think these two comments belong above, where we actually initialize the `stop_pos`.